src/app/store/test-records/reducers/test-records.reducer.ts

```diff
@@ -65,7 +66,20 @@ export const testResultsReducer = createReducer(
     ...state,
     editingTestResult: merge({}, action.testResult)
   })),
-  on(updateResultOfTest, state => ({ ...state, editingTestResult: calculateTestResult(state.editingTestResult) }))
+  on(updateResultOfTest, state => ({ ...state, editingTestResult: calculateTestResult(state.editingTestResult) })),
+  on(updateTestStation, (state, action) => {
+    if (!state.editingTestResult) {
+      return { ...state };
+    }
+    return {
+      ...state,
+      editingTestResult: {
+        ...state.editingTestResult,
+        testStationName: action.payload.testStationName,
+        testStationType: action.payload.testStationType
+      }
+    };
+  })
 );
```

Comment:
Just to make it a bit more concise
```suggestion
  on(updateTestStation, (state, action) => !state.editingTestResult
    ?  { ...state }
    : {
      ...state,
      editingTestResult: {
        ...state.editingTestResult,
        testStationName: action.payload.testStationName,
        testStationType: action.payload.testStationType
      }
    }
);
```

Reply:
Nice